core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java

```diff
@@ -191,24 +192,43 @@ static Expression fromInternal(Expression e, Class<?> targetType) {
     return e;
   }
 
+  /**
+   * build expression list.
+   * if varArgs is true, the last targetType will replace by its Component Type.
+   */
   static List<Expression> fromInternal(Class<?>[] targetTypes,
```

Comment:
Build expression list. If varArgs is true, the last targetType will be replaced by its component type.

Be caution to the character cases.